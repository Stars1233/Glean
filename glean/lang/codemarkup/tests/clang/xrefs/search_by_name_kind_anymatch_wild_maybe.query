query: search.code.SearchByNameKindAndLanguage { name = _, language = Cpp, kinds = { just = _ } }
perf: true
transform: [gensort, []]
