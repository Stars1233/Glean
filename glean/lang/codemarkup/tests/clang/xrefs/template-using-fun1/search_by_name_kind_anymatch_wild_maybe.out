[
  "@generated",
  {
    "key": {
      "name": "N1",
      "language": 0,
      "kinds": 5,
      "entity": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "N1" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 15,
                      "columnEnd": 1
                    }
                  }
                },
                "members": {
                  "key": [
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "N1" } }
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "T" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 11,
                            "columnBegin": 22,
                            "lineEnd": 11,
                            "columnEnd": 33
                          }
                        }
                      }
                    },
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "f" } } },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "N1" } }
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "" },
                                  "type": { "key": "int" }
                                }
                              ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 12,
                            "columnBegin": 1,
                            "lineEnd": 12,
                            "columnEnd": 30
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "N2",
      "language": 0,
      "kinds": 5,
      "entity": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "N2" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 17,
                      "columnBegin": 1,
                      "lineEnd": 25,
                      "columnEnd": 1
                    }
                  }
                },
                "members": {
                  "key": [
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "g" } } },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "N2" } }
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [ ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 20,
                            "columnBegin": 1,
                            "lineEnd": 24,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "N3",
      "language": 0,
      "kinds": 5,
      "entity": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "N3" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 27,
                      "columnBegin": 1,
                      "lineEnd": 36,
                      "columnEnd": 1
                    }
                  }
                },
                "members": {
                  "key": [
                    {
                      "function_": {
                        "key": {
                          "name": {
                            "key": {
                              "name": { "key": { "name": { "key": "g" } } },
                              "scope": {
                                "namespace_": {
                                  "key": { "name": { "key": "N3" } }
                                }
                              }
                            }
                          },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [ ]
                            }
                          },
                          "source": {
                            "file": { "key": "test.cpp" },
                            "lineBegin": 31,
                            "columnBegin": 1,
                            "lineEnd": 35,
                            "columnEnd": 1
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "f",
      "language": 0,
      "kinds": 13,
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "f" } } },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "N1" } } }
                        }
                      }
                    },
                    "signature": {
                      "key": {
                        "returns": { "key": "void" },
                        "parameters": [
                          { "name": { "key": "" }, "type": { "key": "T" } }
                        ]
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 11,
                      "columnBegin": 22,
                      "lineEnd": 11,
                      "columnEnd": 33
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "f",
      "language": 0,
      "kinds": 13,
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "f" } } },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "N1" } } }
                        }
                      }
                    },
                    "signature": {
                      "key": {
                        "returns": { "key": "void" },
                        "parameters": [
                          { "name": { "key": "" }, "type": { "key": "int" } }
                        ]
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 12,
                      "columnBegin": 1,
                      "lineEnd": 12,
                      "columnEnd": 30
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "g",
      "language": 0,
      "kinds": 13,
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "g" } } },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "N2" } } }
                        }
                      }
                    },
                    "signature": {
                      "key": { "returns": { "key": "void" }, "parameters": [ ] }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 20,
                      "columnBegin": 1,
                      "lineEnd": 24,
                      "columnEnd": 1
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "g",
      "language": 0,
      "kinds": 13,
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "g" } } },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "N3" } } }
                        }
                      }
                    },
                    "signature": {
                      "key": { "returns": { "key": "void" }, "parameters": [ ] }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 31,
                      "columnBegin": 1,
                      "lineEnd": 35,
                      "columnEnd": 1
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  }
]